src/emulator/commandUtils.ts

```diff
@@ -156,6 +156,20 @@ export function parseInspectionPort(options: any): number {
   return parsed;
 }
 
+export function shutdownWhenKilled(): Promise<void> {
+  return new Promise((res, rej) => {
+    process.on("SIGINT", async () => {
+      try {
+        await controller.cleanShutdown();
+        res();
+        process.exit(0);
+      } catch (e) {
+        throw new FirebaseError("Failed to shut down cleanly", { original: e, exit: 1 });
```

Comment:
As mentioned above, I don't really think this error will be properly caught since it will only reject the Promise returned by this function, which is unhandled. Best case, Node.js will print out a unhandled promise notice and exit, but I don't think the global error handlers will run at all. Correct me if I'm wrong.

Reply:
Changed how this is done.